src/main/java/com/bite_sized_knowledge/bite_backend/domain/blog/service/BlogService.java

```diff
@@ -0,0 +1,61 @@
+package com.bite_sized_knowledge.bite_backend.domain.blog.service;
+
+import com.bite_sized_knowledge.bite_backend.domain.blog.entity.Blog;
+import com.bite_sized_knowledge.bite_backend.domain.blog.entity.BlogSubscribe;
+import com.bite_sized_knowledge.bite_backend.domain.blog.model.ResBlog;
+import com.bite_sized_knowledge.bite_backend.domain.blog.repository.BlogRepository;
+import com.bite_sized_knowledge.bite_backend.domain.blog.repository.BlogSubscribeRepository;
+import com.bite_sized_knowledge.bite_backend.domain.member.entity.Member;
+import com.bite_sized_knowledge.bite_backend.global.common.error.BadRequestException;
+import jakarta.persistence.EntityManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class BlogService {
+
+    private final BlogRepository blogRepository;
+    private final BlogSubscribeRepository blogSubscribeRepository;
+    private final EntityManager entityManager;
+
+    public ResBlog getBlog(Long blogId, Member member) {
+        validateBlogExists(blogId);
+        Blog blog = entityManager.getReference(Blog.class, blogId);
+        // 구독 여부 조회
+        boolean isSubscribed = blogSubscribeRepository.findByBlogAndMember(blog, member)
+                .filter(BlogSubscribe::isSubscribed)
+                .isPresent();
```

Comment:
제가 잘못 이해했네요, 이렇게 수정하는게 저한테는 조금 더 직관적일 것 같은데, 의견 궁금합니다!
```suggestion
        boolean isSubscribed = blogSubscribeRepository.findByBlogAndMember(blog, member)
                .map(BlogSubscribe::isSubscribed)
                .orElse(false);
```